app/controllers/brexit_checker_controller.rb

```diff
@@ -52,12 +95,47 @@ def confirm_email_signup
 
   def save_results; end
 
+  def saved_results
+    redirect_to transition_checker_new_session_path and return unless logged_in?
+
+    @saved_results =
+      if logged_in?
+        begin
+          Array(
+            update_session_tokens(
+              oidc.get_checker_attribute(
+                access_token: session[:access_token],
+                refresh_token: session[:refresh_token],
+              ),
+            ),
+          )
+        rescue OidcClient::OAuthFailure
+          # this means the refresh token has been revoked or the
+          # accounts services are down
+          logout!
+          []
+        end
+      end
+
+    if @saved_results.empty?
+      redirect_to transition_checker_questions_path
+    else
+      redirect_to transition_checker_results_path(c: @saved_results.last["criteria_keys"])
+    end
+  end
+
```

Comment:
We're only storing one result for now, see https://github.com/alphagov/finder-frontend/pull/2207/files#diff-e4257d6c3bb3ff33ad32620a2d04b871R28

Reply:
Great, I've updated to remove the expectation this will be a hash of arrays.